test/credentials.spec.coffee

```diff
@@ -859,7 +873,7 @@ describe 'AWS.CognitoIdentityCredentials', ->
       expect(creds.cacheId.calls.length).to.equal(0)
       expect(creds.getStorage('id')).not.to.exist
 
-    it 'clears cache if getId fails', ->
```

Comment:
Can you include a corresponding test for each of these scenarios that tests that the cache is cleared for unauthorized users? So one for each of getId, getOpenIdToken, and getCredentialsForIdentity.